superset-frontend/src/views/CRUD/data/dataset/DatasetList.tsx

```diff
@@ -626,6 +628,15 @@ const DatasetList: FunctionComponent<DatasetListProps> = ({
     );
   };
 
+  const location = useLocation();
+  const history = useHistory();
+  useEffect(() => {
```

Comment:
If we are adding the `#create` as a valid URI fragment then I don't think we should automatically remove it. That way if the user refreshes the page, the modal will stay open. We should remove the hash if the user closes the modal. To be consistent, we should also update the hash when the user clicks on `Add dataset` from the list.

Reply:
Modal visibility and `#create` hash now synced!